components/certificate-mgt/org.wso2.carbon.certificate.mgt.core/src/main/java/org/wso2/carbon/certificate/mgt/core/impl/KeyStoreReader.java

```diff
@@ -215,9 +215,14 @@ public Certificate getCertificateByAlias(String alias) throws KeystoreException
         try {
             CertificateManagementDAOFactory.openConnection();
             byte[] certificateBytes = CertificateManagementDAOFactory.getCertificateDAO().retrieveCertificate(alias);
-            raCertificate = (Certificate) Serializer.deserialize(certificateBytes);
+            if (certificateBytes != null) {

```

Comment:
Add a debug log if needed. No need for an error log as we are passing the exception.